treeherder/log_parser/failureline.py

```diff
@@ -94,13 +96,22 @@ def get_kwargs(failure_line):
             if key in failure_line}
 
 
+def create_failure_line(job_log, failure_line):
+    fl = FailureLine.objects.create(repository=job_log.job.repository,
+                                    job_guid=job_log.job.guid,
+                                    job_log=job_log,
+                                    **get_kwargs(failure_line))
+    if "group" in failure_line:
+        # Store as unix path separators and omit the filename
+        group, saved = Group.objects.get_or_create(
+            name=path.dirname(path.join(*failure_line["group"].split("\\"))))
```

Comment:
I would prefer if we could just define an allowed format for these fields and reject any that do not confirm (similar to how we discard jobs that don't conform to the jsonschema schema we define).